src/generators/api-links/index.mjs

```diff
@@ -54,9 +55,11 @@ export default {
     if (input.length > 0) {
       const repositoryDirectory = dirname(input[0].path);
 
-      const repository = getBaseGitHubUrl(repositoryDirectory);
+      const repository = useGit
+        ? getBaseGitHubUrl(repositoryDirectory)
```

Comment:
IMO this is still problematic, ideally we would give a way to specify the correct remote – maybe get it from the NCU config (which would typically be `upstream`, not `origin`)

Reply:
Would you mind opening an issue, explaining the intended behaviour? What is `ncu` in this context btw?

Reply:
NCU is [node-core-utils](https://github.com/nodejs/node-core-utils), the utility that Node.js core collaborators use to e.g. land PRs, start CI, release Node.js. The intended behavior is not assume there's a remove called `origin`

Reply:
Ah that `ncu` -- for some reason I thought of `npm-check-updates` 🤦

Reply:
> NCU is [node-core-utils](https://github.com/nodejs/node-core-utils), the utility that Node.js core collaborators use to e.g. land PRs, start CI, release Node.js. The intended behavior is not assume there's a remove called `origin`

Right, so are you saying the intended approach would be to pass a hash? instead of using git? Or what exactly are you thinking we could pass on the CLI? 🤔

Reply:
I didn't think about that, but yeah, it's probably a better idea to ask the user to provide the repo URL (or owner/name if we're fine assuming it's hosted on github.com) – alternatively, the user could provide a remote name and we keep the same algorithm, but IMO that's a worse UX.  We could also ask to provide a hash, although I don't find the `git rev-parse` very problematic as long as it's opt-in.

Reply:
What if the CLI arg (the opt-in one) asks to pass the whole patch url?

ie: https://github.com/nodejs/node/commit/3861eccc05751eea2a0915316c16b86caea531ba.patch or https://github.com/nodejs/node/commit/3861eccc05751eea2a0915316c16b86caea531ba

from there we can infer the whole ordeal just from the URL, no?

Reply:
Sure, that's fine by me as long as it doesn't assume a `origin` remote

Reply:
Wdym by assuming an origin remote? Are we stamping/storing somewhere a remote name?

Reply:
Yes, that’s the whole issue we’ve been discussing! It’s here: https://github.com/nodejs/api-docs-tooling/blob/c6e8099db8c6f2b4d31692172e6a661102e91dee/src/generators/api-links/utils/getBaseGitHubUrl.mjs#L9

Reply:
Ah yes, with the cli flag we would not be using any of these commands :)

Reply:
@aduh95 done!